bukuserver/response.py

```diff
@@ -1,4 +1,45 @@
-response_template = {
-    "success": {'status': 0, 'message': 'success'},
-    "failure": {'status': 1, 'message': 'failure'}
-}
+from enum import Enum
+from flask import jsonify
+from flask_api import status
+
+
+class Response(Enum):
+    SUCCESS = (status.HTTP_200_OK, "Success.")
+    FAILURE = (status.HTTP_400_BAD_REQUEST, "Failure.")
+    INPUT_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Request not valid")
+    JSON_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Failed to decode JSON data.")
+    BOOKMARK_NOT_FOUND = (status.HTTP_404_NOT_FOUND, "Bookmark not found.")
+    TAGS_MISSING = (status.HTTP_400_BAD_REQUEST, "Replacement tags are missing.")
+    TAG_NOT_FOUND = (status.HTTP_404_NOT_FOUND, "Tag not found.")
+    TAGS_NOT_LIST = (status.HTTP_400_BAD_REQUEST, "List of tags expected.")
+    TAG_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Invalid tag.")
+
+    @property
+    def status_code(self):
+        return self.value[0]
+
+    @property
+    def exit_code(self):
+        return int(self.status_code != status.HTTP_200_OK)
```

Comment:
I'd say that the "boolean is implemented as `0`/`1` alias" thing is an implementation detail of CPython, and shouldn't be relied on in application logic.

It's better to define something like `OK, FAIL = 1, 0` at the top of the file and return those values explicitly by name (e.g. via `if`/`else`).

…Also, "exit code" is a process execution term – it hardly fits here as there's nothing being "exited" (and this is basically an oddly presented boolean, so calling it a "code" is also kind of a stretch). Maybe call it just `status` instead? (Thus making it so that "status" is `1` or `0`, and "status code" is `200`, `400`, `404`…)